### DISCUSSION TITLE
PiFace Shim RTC - Support mit aktueller RaspberryMatic?

### DISCUSSION THREAD

**Discussion by Baxxy13** [OpenCCU/OpenCCU#1880] [🇩🇪 Fragen&Antworten] [answered]
Vor langer Zeit wurde ja mal der Support für die PiFace Shim RTC in RaspberryMatic integriert.
[RaspberryMatic - RealTimeClock (RTC) Installation](https://homematic-forum.de/forum/viewtopic.php?f=65&t=27684&sid=e4d4c4ff5947dfaa1abdbe164ed92508#p245005)
Das hatte ich Anfangs auch gerne genutzt da die alten Funkmodule noch keine RTC hatten. 
Mit dem Wechsel auf RPI-RF-MOD's verschwanden die ShimRTC's dann in der Bastelkiste.

Für ein paar Testereien hatte ich gestern einen Pi3B (**RM 3.63.9.20220625**) mit der ShimRTC und ohne Funkmodul aufgesetzt.
Dabei sind mir ein paar ungewöhnliche Logeinträge aufgefallen.
```
Jul 17 00:52:04 localhost kern.warn kernel: [   10.373482] rtc-ds1307: probe of 1-0032 failed with error -121
Jul 17 00:52:04 localhost kern.info kernel: [   10.378520] rtc-ds1307 1-006f: registered as rtc0
Jul 17 00:52:04 localhost kern.info kernel: [   11.159621] pl011_raw_uart 3f201000.serial: Missing I2C driver of rtc device, trying to load
Jul 17 00:52:04 localhost kern.warn kernel: [   11.173645] rtc-ds1307: probe of 1-0032 failed with error -121
Jul 17 00:52:30 homematic-raspi daemon.info chronyd[521]: chronyd version 4.2 starting (+CMDMON +NTP +REFCLOCK +RTC -PRIVDROP -SCFILTER -SIGND +ASYNCDNS -NTS -SECHASH +IPV6 -DEBUG)
Jul 17 01:00:00 homematic-raspi kern.err kernel: [  484.643577] rtc-ds1307 1-006f: write error -121
Jul 17 01:00:00 homematic-raspi kern.err kernel: [  484.643962] rtc-ds1307 1-006f: read error -121
Jul 17 01:11:00 homematic-raspi kern.err kernel: [ 1144.675391] rtc-ds1307 1-006f: write error -121
Jul 17 01:11:00 homematic-raspi kern.err kernel: [ 1144.675798] rtc-ds1307 1-006f: read error -121
Jul 17 01:22:00 homematic-raspi kern.err kernel: [ 1804.705629] rtc-ds1307 1-006f: write error -121
Jul 17 01:22:00 homematic-raspi kern.err kernel: [ 1804.706004] rtc-ds1307 1-006f: read error -121
Jul 17 01:33:00 homematic-raspi kern.err kernel: [ 2464.735435] rtc-ds1307 1-006f: write error -121
Jul 17 01:33:00 homematic-raspi kern.err kernel: [ 2464.735812] rtc-ds1307 1-006f: read error -121
```
Zum einen der "missing driver".
Zum anderen wird ja nach der RTC eines RPI-RF-MOD auf Adresse 32 gescannt. 
"Komisch" ist hierbei das da "rtc-ds1307" steht obwohl das RPI-RF-MOD eine "rtc-rx8130" hat.

Spaßenshalber zurück auf die gute alte RM **3.51.6.20200621** sah das noch so aus:
```
Jul 17 10:11:15 localhost user.info kernel: [    8.596596] rtc-ds1307 1-006f: registered as rtc0
Jul 17 10:11:15 localhost user.err kernel: [    9.476760] rtc-rx8130 1-0032: Unable to read registers #28..#30
Jul 17 10:11:15 localhost user.warn kernel: [    9.476797] rtc-rx8130: probe of 1-0032 failed with error -121
Jul 17 10:11:49 homematic-raspi daemon.info chronyd[562]: chronyd version 3.5 starting (+CMDMON +NTP +REFCLOCK +RTC -PRIVDROP -SCFILTER -SIGND +ASYNCDNS -SECHASH +IPV6 -DEBUG)
```
Hier sieht alles gut aus.
Keine "missing driver" Info.
ShimRTC (rtc-ds1307 an 6f) läuft
Scan nach RPI-RF-MOD RTC (rtc-rx8130 an 32) schlägt erwartungsgemäß fehl.

Zusatz:
Auf einem Pi3B+ (**RM 3.63.9.20220625**) mit RPI-RF-MOD sieht das so aus:
`Jul 17 09:49:06 localhost kern.info kernel: [   10.010027] rtc-ds1307 1-0032: registered as rtc0`

### Edit:
Eher doch viel Wind um nichts.
Nachdem ich die ShimRTC nochmal neu draufgesetzt hatte sind nun die read/write errors auch in der aktuellen RM weg.

Was bleibt sind 2 Fragen:
- die "driver missing" Info, ist die relevant?
- warum wird die RTC des RPI-RF-MOD (rtc-rx8130) in aktuellen Versionen als rtc-ds1307 deklariert? Das war ja in älteren Versionen noch nicht so.

Danke für's lesen! ;-)

**Comment by jens-maus** [comment, marked as answer]
> Was bleibt sind 2 Fragen:
> 
> * die "driver missing" Info, ist die relevant?

Nein, das sollte nicht relevant sein.

> * warum wird die RTC des RPI-RF-MOD (rtc-rx8130) in aktuellen Versionen als rtc-ds1307 deklariert? Das war ja in älteren Versionen noch nicht so.

Weil es ab Linux kernel 5.15.x eine Änderung bzw. direkte Implementation der rx8130 RTC gab und diese Implementation teil des größeren `rtc-ds1307` kernel modul wurde. In diesem sind noch andere RTC Varianten untergebracht und nun eben auch das `rx8130` das es früher nur als externes, selbstkompilierbares kernel modul gab, aber nun eben seit Kernel 5.15.x teil des Linux Kernels geworden ist.
